### PR TITLE
Remove TypeInfo

### PR DESCRIPTION
**Pull request by iamcarbon**
Replaces TypeInfo with Type.

### CODE REVIEW

File: src/Peachpie.Runtime/Reflection/TypeFields.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public class TypeFields
 
         internal TypeFields(Type tinfo)
         {
-            _fields = tinfo.GetFields(BindingFlags.Instance | BindingFlags.DeclaredOnly).Where(_IsAllowedField).ToDictionary(_FieldName, StringComparer.Ordinal);
+            _fields = tinfo.GetFields().Where(_IsAllowedField).ToDictionary(_FieldName, StringComparer.Ordinal);
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/TypeFields.cs]: as I know, this has a different semantic - we MUST collect only declared fields (not fields from base type)

File: src/Peachpie.Runtime/Reflection/TypeFields.cs [hunk 2]
```diff
@@ -56,10 +56,11 @@ internal TypeFields(Type tinfo)
                     staticscontainer = staticscontainer.MakeGenericType(tinfo.GenericTypeArguments);
                 }
 
-                _staticsFields = staticscontainer.GetFields(BindingFlags.Instance | BindingFlags.DeclaredOnly).ToDictionary(_FieldName, StringComparer.Ordinal);
+                _staticsFields = staticscontainer.GetFields().ToDictionary(_FieldName, StringComparer.Ordinal);
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/TypeFields.cs]: as above

File: src/Peachpie.Runtime/Reflection/TypeFields.cs [hunk 3]
```diff
@@ -56,10 +56,11 @@ internal TypeFields(Type tinfo)
...
             }
 
-            _properties = tinfo.GetProperties(BindingFlags.Instance | BindingFlags.DeclaredOnly).Where(_IsAllowedProperty).ToDictionary(_PropertyName, StringComparer.Ordinal);
+            _properties = tinfo.GetProperties().Where(_IsAllowedProperty).ToDictionary(_PropertyName, StringComparer.Ordinal);
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/TypeFields.cs]: as above

File: src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs [hunk 4]
```diff
@@ -205,9 +199,9 @@ TObjectCreator BuildCreatorProtected()
                 {
                     var ctorsList = new List<ConstructorInfo>();
                     bool hasProtected = false;
-                    foreach (var c in _type.GetConstructors(BindingFlags.DeclaredOnly))
+                    foreach (var c in _type.GetConstructors(BindingFlags.Instance | BindingFlags.Public))
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs]: this breaks the code - we need all no private ctors

File: src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs [hunk 5]
```diff
@@ -177,9 +171,9 @@ TObjectCreator BuildCreatorPrivate()
                 {
                     var ctorsList = new List<ConstructorInfo>();
                     bool hasPrivate = false;
-                    foreach (var c in _type.GetConstructors(BindingFlags.DeclaredOnly))
+                    foreach (var c in _type.GetConstructors(BindingFlags.Instance))
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs]: + `DeclaredOnly` on current type only, not base types !!!

File: src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs (hunk 4 above)
Comment (jakubmisek) [inline src/Peachpie.Runtime/Reflection/PhpTypeInfo.cs]: not base types, `DeclaredOnly` !!

File: src/Peachpie.Runtime/Dynamic/Cache.cs [hunk 6]
```diff
@@ -123,46 +119,19 @@ public static class Object
...
-                {
-                    if (Enumerable.SequenceEqual(ptypes, ps.Select(p => p.ParameterType)))
-                    {
-                        return ctor;
-                    }
-                }
-            }
-
-            throw new ArgumentException();
+            return type.GetConstructor(ptypes) ?? throw new ArgumentException();
```
Comment (jakubmisek) [inline src/Peachpie.Runtime/Dynamic/Cache.cs]: `DeclaredOnly`

**Comment by jakubmisek** [comment]
These commits change internal reflection significantly, used method calls have different semantics. I also think. there is no reason for these modifications.

**Comment by iamcarbon** [comment]
Agreed. Backing out of this change.

**Comment by jakubmisek** [comment]
I'm always little bit scared when so many magic code is being changed at once : )

**Pull request closed without merging**